### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Monoidal): Day's reflection theorem

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Yoneda.lean [hunk 1]
```diff
@@ -437,6 +437,19 @@ lemma isIso_of_yoneda_map_bijective {X Y : C} (f : X ⟶ Y)
   obtain ⟨g, hg : g ≫ f = 𝟙 Y⟩ := (hf Y).2 (𝟙 Y)
   exact ⟨g, (hf _).1 (by aesop_cat), hg⟩
 
+lemma isIso_iff_isIso_yoneda_map {X Y : C} (f : X ⟶ Y) :
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Yoneda.lean:440]: I would suggest adding first a variant `isIso_iff_yoneda_map_bijective`, and then deducing this one.

File: Mathlib/CategoryTheory/Monoidal/Braided/Reflection.lean [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    simp
+  tfae_finish
+
+end
+
+section
+variable [MonoidalCategory C]
+variable {L : MonoidalFunctor D C} {R : C ⥤ D} [R.Faithful] [R.Full] (adj : L.toFunctor ⊣ R)
+
+include adj in
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Monoidal/Braided/Reflection.lean:194]: Do you need this? After all `adj` is explicitly mentioned in the statement. Idem below.

File: Mathlib/CategoryTheory/Monoidal/Braided/Reflection.lean [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+import Mathlib.CategoryTheory.Closed.Monoidal
+import Mathlib.CategoryTheory.Monad.Adjunction
+import Mathlib.CategoryTheory.Monoidal.Braided.Basic
+import Mathlib.Tactic.TFAE
+/-!
+
+# Day's reflection theorem
+
+Let `D` be a symmetric monoidal closed category and let `C` be a reflective subcategory. Day's
+reflection theorem gives four equivalent conditions, which are all of the form that a map obtained
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Monoidal/Braided/Reflection.lean:16]:
```suggestion
reflection theorem proves the equivalence of four conditions, which are all of the form that a map obtained
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by dagurtomas** [comment]
Thanks!

bors merge

**Comment by dagurtomas** [comment]
bors merge

**Pull request closed without merging**